django/db/models/fields/related.py

```diff
@@ -2101,9 +2139,22 @@ class OneToOneField(ForeignKey):
 
     description = _("One-to-one relationship")
 
-    def __init__(self, to, to_field=None, **kwargs):
+    def __init__(self, to, on_delete=None, to_field=None, **kwargs):
         kwargs['unique'] = True
-        super(OneToOneField, self).__init__(to, to_field, **kwargs)
+
+        if on_delete is not None and not callable(on_delete):
+            warnings.warn(
```

Comment:
same as above. The only way I can see to hit this warning would be pass as args in opposite order: `models.OneToOneField(Question, 'blah', models.PROTECT)`. Is that a case we need to guard against?